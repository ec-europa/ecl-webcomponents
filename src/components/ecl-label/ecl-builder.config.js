const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-label-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-label-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-label-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-label-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
