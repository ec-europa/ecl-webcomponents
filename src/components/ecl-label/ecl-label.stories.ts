const getArgs = () => {
  return {
    variant: 'low',
    label: 'Label',
  };
};

const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: ['low', 'medium', 'high', 'highlight'],
      description: 'Label variant',
    },
    label: {
      type: { name: 'string' },
      description: 'Label to be shown',
    },
  };
};

export default {
  title: 'Components/label',
};

const Template = args =>
`<ecl-label
  variant="${args.variant}"
  theme="${args.theme}"
>
  ${args.label}
</ecl-label>`;

export const Label = Template.bind({});
Label.storyName = 'default';
Label.args = getArgs();
Label.argTypes = getArgTypes();
