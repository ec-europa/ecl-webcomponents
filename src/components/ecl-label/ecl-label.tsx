import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-label',
  styleUrls: {
    ec: './build/styles/ecl-label-ec.css',
    eu: './build/styles/ecl-label-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclLabel {
  @Prop() styleClass: string = '';
  @Prop() variant: string = 'low';
  @Prop() theme: string = 'ec';

  getClass(): string {
    return [
      `ecl-label`,
      `ecl-label--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <span class={this.getClass()}>
        <slot></slot>
      </span>
    )
  }
}
