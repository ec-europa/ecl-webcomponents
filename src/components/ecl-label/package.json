{
  "name": "@ecl/web-components-label",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.1.0",
  "description": "ECL Label",
  "main": "ecl-label.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env NODE_ENV=production ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.7.1",
    "@ecl/vanilla-component-label": "3.7.1",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
