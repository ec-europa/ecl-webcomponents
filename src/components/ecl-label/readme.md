# ecl-label



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default |
| ------------ | ------------- | ----------- | -------- | ------- |
| `styleClass` | `style-class` |             | `string` | `''`    |
| `theme`      | `theme`       |             | `string` | `'ec'`  |
| `variant`    | `variant`     |             | `string` | `'low'` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
