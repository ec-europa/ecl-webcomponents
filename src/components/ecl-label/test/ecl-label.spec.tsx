import { newSpecPage } from '@stencil/core/testing';
import { EclLabel } from '../ecl-label';

describe('ecl-label', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclLabel],
      html: `<ecl-label
              variant="low"
              theme="ec"
            >
              Label
            </ecl-label>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
