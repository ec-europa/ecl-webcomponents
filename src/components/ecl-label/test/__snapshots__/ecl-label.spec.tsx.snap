// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-label renders 1`] = `
<ecl-label theme="ec" variant="low">
  <!---->
  <span class="ecl-label ecl-label--low">
    Label
  </span>
</ecl-label>
`;
